src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/RenamerProjectTreeActionHandler.cs

```diff
@@ -143,7 +153,10 @@ await base.RenameAsync(context, node, value); // Async process (1 of which updat
                 //
                 // Because HandleRename() is called after Roslyn, we can set this 
                 // flag when node.FilePath == HandlRename(FilePath)
-                while (_isLanguageServiceDone == false) ;
+                if (_languageServiceDone.TryGetValue((oldFilePath, newFileNameWithExtension), out SemaphoreSlim semLanguegeServiceDone)) {
```

Comment:
This needs to be thread-safe; HandleRename can overlap with this and corrupt the dictionary.

Reply:
done in next commit.